std/experimental/ndslice/slice.d

```diff
@@ -57,7 +57,7 @@ auto sliced(
 auto sliced(
     Flag!"replaceArrayWithPointer" replaceArrayWithPointer = Yes.replaceArrayWithPointer,
     Flag!"allowDownsize" allowDownsize = No.allowDownsize,
-    size_t N, Range)(Range range, auto ref in size_t[N] lengths, size_t shift = 0)
+    size_t N, Range)(Range range, auto ref in size_t[N] lengths, size_t shift = 0) @trusted
```

Comment:
I don't like this. This function is too big and complicated to be marked `@trusted`.  Can you narrow down the parts that need to be `@trusted` and refactor them into trusted lambdas instead? Especially since this is a template function and we should be letting the compiler infer `@safe` instead of putting blind faith in the correctness of the code with `@trusted`.